### DISCUSSION TITLE
Operator corrupted

### DISCUSSION THREAD

**Discussion by marcelloraffaele** [rabbitmq/cluster-operator#948] [Q&A] [answered]
Does someone know a way to clean up the resources and reinstall the rabbitmq operator from the beginning?
I am trying to reinstall the operator but the namespace `rabbitmq-system` remains in `terminating state` and if we force the delation, the installation doesn't work. It seems that some old resource is blocking the installation.

**Comment by Zerpet** [comment, marked as answer]
If you deployed any `RabbitmqCluster` object in `rabbitmq-system` namespace, those might be there, terminating, blocked by a finalizer. You can check this by running `kubectl -n rabbitmq-system get rmq`. This can happen if you delete the namespace, with rabbit objects in it, because the Operator Pod exits before it can remove the finalizers from the rabbit objects. The same applies to objects created by the Messaging Topology Operator.

If you have any `RabbitmqCluster` objects left, you can edit `kubectl edit rmq <name>` and remove the `finalizer` section. That should allow the deletion to proceed. Note that the namespace object might still need some time (2-3 minutes) after all leftover objects are deleted.

**Comment by marcelloraffaele** [comment]
Thank you for the answer, but what if the operator itself is corrupted and I am not able to delete it and clean the environment?

**Reply by Zerpet** [reply]
The Operator itself doesn't have a finalizer, or anything that would block its deletion. You'd likely have to look in the Kubernetes node hosting the container for issues. The kubelet logs may have more information.
